### PR TITLE
refactor: removed async module usage

### PR DESCRIPTION
**Pull request by vijay-qlogic**
Fixes [#2883](https://github.com/googleapis/google-cloud-node/issues/2883) (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/textDetection.js [hunk 1]
```diff
@@ -15,10 +15,9 @@
 
 'use strict';
 
-const async = require('async');
 const fs = require('fs');
 const path = require('path');
-
+const {promisify} = require('util');
```
Comment (romlogic) [inline samples/textDetection.js]: line 24. the link is not working, and please add a full stop (period) for all the comments with complete sentences as some of them have full stops (line 33), and others don't.

File: samples/textDetection.js [hunk 2]
```diff
@@ -89,7 +88,7 @@ Index.prototype.lookup = function(words, callback) {
       self.tokenClient.smembers(word, cb);
     };
   });
-  async.parallel(tasks, callback);
+  Promise.all(tasks).then(callback);
```
Comment (AVaksman) [inline samples/textDetection.js]: Same comment here, I think you can just `return Promise.all(taskss);`
and line 83 make into
`Index.prototype.lookup = async words => {`

File: samples/textDetection.js [hunk 3]
```diff
@@ -78,7 +77,7 @@ Index.prototype.add = function(filename, document, callback) {
     self.tokenClient.set(filename, document, cb);
   });
 
-  async.parallel(tasks, callback);
+  Promise.all(tasks).then(callback);
```
Comment (AVaksman) [inline samples/textDetection.js]: I think you can just ```return Promise.all(tasks);``` here 
and on line 60 make it into an async function
`Index.prototype.add = async (filename, document) => {`

File: samples/textDetection.js (hunk 2 above)
Comment (vijay-qlogic) [reply samples/textDetection.js]: Done! https://github.com/vijay-qlogic/nodejs-vision/pull/1/commits/bb6ccacc6481592a693e8d7c357219f88de87c96

File: samples/textDetection.js (hunk 3 above)
Comment (vijay-qlogic) [reply samples/textDetection.js]: Done ! 
https://github.com/vijay-qlogic/nodejs-vision/pull/1/commits/bb6ccacc6481592a693e8d7c357219f88de87c96

File: samples/textDetection.js (hunk 1 above)
Comment (vijay-qlogic) [reply samples/textDetection.js]: Done! 
comments from line-21 to line-23 is single sentence, so not added full stop (period) there, elsewhere it's added.
link update to https://cloud.google.com/docs/authentication/getting-started, this page speaks as per the comments.

https://github.com/vijay-qlogic/nodejs-vision/pull/1/commits/bb6ccacc6481592a693e8d7c357219f88de87c96

**Comment by AVaksman** [comment]
for `getTextFromFiles()` function:
refactor to 
```javascript
async function getTextFromFiles(index, inputFiles) {
  // Make a call to the Vision API to detect text
  const requests = inputFiles.map(filename => {
    return {
      image: {content: fs.readFileSync(filename).toString('base64')},
      features: [{type: 'TEXT_DETECTION'}],
    };
  });
  const [results] = await client.batchAnnotateImages({requests: requests});
  const detections = results[0].responses;
  const textResponse = {};
  const tasks = inputFiles.map((filename, i) => {
    const response = detections[i];
    if (response.error) {
      console.log('API Error for ' + filename, response.error);
      return;
    } else if (Array.isArray(response)) {
      textResponse[filename] = 1;
    } else {
      textResponse[filename] = 0;
    }
    return cb => extractDescriptions(filename, index, response, cb);
  });
  return Promise.all(tasks);
}
```

**Comment by AVaksman** [comment]
refactor `main` function to
```javascript
// Run the example
async function main(inputDir) {
  const index = new Index();
  // Scan the specified directory for files
  const files = fs.readdirSync(inputDir);
  // Separate directories from files
  const allImageFiles = files.map(file => {
    const filename = path.join(inputDir, file);
    if (!fs.statSync(filename).isDirectory()) {
      return filename;
    }
  });
  // Figure out which files have already been processed
  const tasks = allImageFiles
    .filter(filename => filename)
    .map(filename => {
      return cb => {
        index.documentIsProcessed(filename, function(err, processed) {
          if (err) {
            return cb(err);
          }
          if (!processed) {
            // Forward this filename on for further processing
            return cb(null, filename);
          }
          cb();
        });
      };
    });
  // Analyze any remaining unprocessed files
  const imageFilesToProcess = await Promise.all(tasks).filter(
    filename => filename
  );
  if (imageFilesToProcess.length) {
    return getTextFromFiles(index, imageFilesToProcess);
  }
  console.log('All files processed!');
  return Promise.resolve();
}
```

File: samples/textDetection.js [hunk 4]
```diff
@@ -15,19 +15,18 @@
 
 'use strict';
 
-const async = require('async');
 const fs = require('fs');
 const path = require('path');
-
+const {promisify} = require('util');
```
Comment (AVaksman) [inline samples/textDetection.js]: after proposed refactoring `promisify`  is not needed any more, please remove it.

Comment (vijay-qlogic) [reply samples/textDetection.js]: Done!

**Comment by AVaksman** [comment]
This task was completed by JustinBeckwith as part of a bigger PR.
#297

**Comment by vijay-qlogic** [comment]
Alright! Closing this PR then.

**Pull request closed without merging**